resolves #2028 : work on add heatfluid toolbox python wrapper

prudhomm opened the pull request:
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/feelpp/feelpp/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes?
- [x] Have you successfully run the Feel++ testsuite with your changes locally?
- [ ] Have you written Doxygen comments in your contribution ?

-----

- closes #2028

prudhomm committed:
work on add heatfluid toolbox python wrapper #2028

- add initial implementation
- not tested at all but compiles

/cc @thomas-saigre

prudhomm committed:
up heat and fluid interface for heatfluid wrapper

- update __init__.py to generate proper function pointers

add heatfluid toolbox python wrapper #2028

/cc @thomas-saigre
skip feelpp skip tests skip toolboxes skip mor

prudhomm committed:
do not install

add heatfluid toolbox python wrapper #2028

prudhomm committed:
do not install

add heatfluid toolbox python wrapper #2028

skip feelpp skip tests skip toolboxes skip mor

prudhomm committed:
skip feelpp skip tests skip toolboxes skip mor

prudhomm committed:
Merge branch '2028-add-heatfluid-toolbox-python-wrapper' of https://github.com/feelpp/feelpp into 2028-add-heatfluid-toolbox-python-wrapper

skip feelpp skip tests skip toolboxes skip mor

prudhomm committed:
add heatfluid toolbox python wrapper fixes #2028

- fix init.py for heatfluid and thermoelectric
- add tests for heatfluid
- add tests for thermoelectric

/cc @thomas-saigre
skip feelpp skip tests skip toolboxes skip mor

prudhomm commented:
@thomas-saigre @vincentchabannes the tests for heatfluid and thermoelectric are up and running
let's see what the CI says

prudhomm committed:
up toolboxe tests

skip feelpp skip toolboxes skip tests skip mor

prudhomm commented:
@vincentchabannes the pr is ready

vincentchabannes commented on python/pyfeelpp-toolboxes/feelpp/toolboxes/heatfluid/heatfluid.py:
```diff
@@ -0,0 +1,11 @@
+import sys
+import feelpp
+from feelpp.toolboxes.thermoelectric import *
+e=feelpp.Environment(sys.argv,opts=toolboxes_options("thermo-electric"))
```
thermo-electric -> heat-fluid

prudhomm replied:
I will remove the code heatfluid.py it should not be there.

vincentchabannes commented on python/pyfeelpp-toolboxes/feelpp/toolboxes/heatfluid/heatfluid.py:
```diff
@@ -0,0 +1,11 @@
+e=feelpp.Environment(sys.argv,opts=toolboxes_options("thermo-electric"))
+
+
+f=thermoelectric(dim=2,orderPotential=1,worldComm=e.worldCommPtr())
```
heatfluid and modifiy arguments

prudhomm replied:
I will remove the code `heatfluid.py` it should not be there.

vincentchabannes commented on python/pyfeelpp-toolboxes/feelpp/toolboxes/heatfluid/__init__.py:
```diff
@@ -0,0 +1,38 @@
+    """create a heatfluid toolbox solver
+    Keyword arguments:
+    dim -- the dimension (default: 2)
+    orderPotential -- the polynomial order for the potential (default: 1)
```
fix copy past for heatfluid

vincentchabannes commented on python/pyfeelpp-toolboxes/feelpp/toolboxes/heatfluid/heatfluid.cpp:
```diff
@@ -0,0 +1,103 @@
+    py::class_<toolbox_t,std::shared_ptr<toolbox_t>,ModelNumerical>(m,pyclass_name.c_str())
+        .def(py::init<std::string const&,std::string const&,worldcomm_ptr_t const&,std::string const&, ModelBaseRepository const&>(),
+             py::arg("prefix"),
+             py::arg("keyword")=std::string("thermo-electric"),
```
heat-fluid

prudhomm replied:
it is done properly (passing keyword as `heat-fluid` by default in `__init__,py` but indeed this is not nice

prudhomm committed:
add heatfluid toolbox python wrapper #2028

fix heat fluid keyword @vincentchabannes

skip feelpp skip tests skip toolboxes skip mor

prudhomm committed:
up docstring

[ci skip] [skip ci]

prudhomm committed:
rm file, fix install

add heatfluid toolbox python wrapper #2028

[ci skip]
